firmware/library/L1_Drivers/onboardled.hpp

```diff
@@ -7,14 +7,96 @@
 #pragma once
 
 #include <cstdint>
+#include "L1_Drivers/gpio.hpp"
+#include "L2_Utilities/macros.hpp"
 
 class OnBoardLedInterface {
  public:
-           virtual void Initialize()                                 = 0;
-           virtual void LedOn(uint8_t led_number)                    = 0;
-           virtual void LedOff(uint8_t led_number)                   = 0;
-           virtual void LedSet(uint8_t led_number, bool on)          = 0;
-           virtual void LedToggle(uint8_t led_number)                = 0;
-           virtual void LedSetAll(uint8_t value)                     = 0;
-           virtual uint8_t LedGetValues(void)                        = 0;
+  enum LightState : bool
+  {
+  kOn = true,
+  kOff = false
+  };
+           virtual void Initialize()                                      = 0;
+           virtual void On(uint8_t led_number)                            = 0;
+           virtual void Off(uint8_t led_number)                           = 0;
+           virtual void Set(uint8_t led_number,
+              OnBoardLedInterface::LightState state)                      = 0;
+           virtual void Toggle(uint8_t led_number)                        = 0;
+           virtual void SetAll(uint8_t value)                             = 0;
+           virtual uint8_t GetStates(void)                                = 0;
+};
+
+class OnBoardLed: public OnBoardLedInterface
+{
+ public:
+  // Initialize takes the array of Gpios, sets each one to an output, and
+  // then turns all of the leds by setting the output high.
+  void Initialize(void) override
+    {
+      for (uint8_t i = 0; i < 4; i++)
+      {
+        led[i].SetAsOutput();
+        led[i].SetHigh();
+      }
+    }
+
+  void On(uint8_t led_number) override
+  {
+    if (led_number > 3) {printf("Input Led number can't be greater than 3.");}
```

Comment:
Rather than running your own printf case evaluation use the SJ2_ASSERT_WARNING(led < 4, "Input Led number can't be greater than 3, input = %d.", led_number);

Reply:
That macro can be found in the L2/Macros.hpp file. Also, do the same for the other methods.

Reply:
This has been fixed.